oras/oci.py

```diff
@@ -44,22 +44,43 @@ def get_annotations(self, section: str) -> dict:
         return {}
 
 
-def NewLayer(blob, media_type: str = None, is_dir: bool = False) -> dict:
+class Layer:
+    def __init__(
+        self, blob: str, media_type: Optional[str] = None, is_dir: bool = False
```

Comment:
just to confirm -- blob is a path here? I think it could be confusing to users wether blob is a path to a file or a string to upload.
If we'd use `pathlib.Path` as type, it would probably be pretty obvious.

However, we could also use a `io` type as input (so that the user can give a File, StringIO, or BytesIO object).

Just some additional thoughts. 

We could also have named functions `Layer.from_string(...)`, `Layer.from_blob(BytesIO(...))` etc.